src/server/hook_func.c

```diff
@@ -3676,6 +3684,95 @@ write_hook_accept_debug_output_and_close(void)
 	}
 }
 
+/**
+ * @brief
+ * 		getting the vnode attributes and resource list
+ *			each node attribute will be of the format:
+ *			<node_name>.<attr_name>
+ *
+ * @see
+ * 		run_periodic_hook
+ *
+ * @return void
+ */
+static pbs_list_head *
+get_vnode_list(void){
+	int i;
+	int index;
+	char name_str_buf[STRBUF + 1] = {'\0'};
+	struct pbsnode	*pnode = NULL;
+	attribute_def *padef = node_attr_def;
+
+	CLEAR_HEAD(vnode_attr_list);
+	for (i = 0; i < svr_totnodes; i++) {
+		pnode = pbsndlist[i];
+		for (index = 0; index < ND_ATR_LAST; index++) {
+			if ((padef+index)->at_flags & ATR_VFLAG_SET) {
+				strncpy(name_str_buf, pnode->nd_name, STRBUF);
+				strcat(name_str_buf, ".");
+				strncat(name_str_buf, (padef+index)->at_name, (STRBUF - strlen(name_str_buf)));
+				if ((padef+index)->at_encode(
+					&pnode->nd_attr[index],
+					&vnode_attr_list, name_str_buf,
+					(char *)0, ATR_ENCODE_HOOK, NULL) < 0) {
+					snprintf(log_buffer, sizeof(log_buffer),
+						"error on encoding node %s attributes!",
```

Comment:
for debugging purpose also mention attribute name in the logs

Reply:
Okay. Wrote name_str_buf instead as it has both node and attr info.